tests/PHPStan/Rules/Functions/CallToFunctionParametersRuleTest.php

```diff
@@ -1646,4 +1646,38 @@ public function testArgon2PasswordHash(): void
 		$this->analyse([__DIR__ . '/data/argon2id-password-hash.php'], []);
 	}
 
+	public function testBugInstanceofStaticVsThis(): void
+	{
+		if (PHP_VERSION_ID < 80100) {
+			$this->markTestSkipped('Test requires PHP 8.1.');
+		}
+
+		$this->analyse([__DIR__ . '/../Properties/data/bug-instanceof-static-vs-this-property-assign.php'], [
+			[
+				'Parameter #1 $var is passed by reference so it does not accept readonly property BugInstanceofStaticVsThisPropertyAssign\FooChild::$nativeReadonlyProp.',
```

Comment:
Unrelated to this PR: It should probably complain about `@readonly` properties as well.